theseus/base/utilities/download.py

```diff
@@ -60,18 +62,27 @@ def download_from_url(url, root=None, filename=None):
     return fpath
 
 
-def download_from_wandb(filename, run_path, save_dir, generate_id_text_file=False):
-    import wandb
+def download_from_wandb(filename, run_path, save_dir, rename=None, generate_id_text_file=False):
 
+    import wandb
+    
     try:
         path = wandb.restore(filename, run_path=run_path, root=save_dir)
-
+        
         # Save run id to wandb_id.txt
         if generate_id_text_file:
             wandb_id = osp.basename(run_path)
             with open(osp.join(save_dir, "wandb_id.txt"), "w") as f:
                 f.write(wandb_id)
-
+        
+        if rename:
+            new_name = str(Path(path.name).parent / rename)
+            os.rename(path.name, new_name)
+            return new_name
```

Comment:
this line would break out of the function and skip all the below lines, wouldn't it? Should have overridden the `path` variable instead

Reply:
Silly me! I fixed it ^^. Thank you.

Reply:
As a side note, path.name is not writable.